src/components/Tree/TreeListItem.tsx

```diff
@@ -0,0 +1,156 @@
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+
+import Tree from './Tree'
+import TreeTitle from './TreeTitle'
+import { defaultBehavior } from '../../lib/accessibility'
+import { Accessibility } from '../../lib/accessibility/types'
+
+import {
+  AutoControlledComponent,
+  childrenExist,
+  customPropTypes,
+  createShorthandFactory,
+} from '../../lib'
+import {
+  ComponentEventHandler,
+  ShorthandRenderFunction,
+  ShorthandValue,
+} from '../../../types/utils'
+import {
+  commonUIComponentPropTypes,
+  childrenComponentPropTypes,
+  contentComponentPropsTypes,
+} from '../../lib/commonPropTypes'
+import {
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+} from '../../lib/commonPropInterfaces'
+
+export interface TreeListItemProps
+  extends UIComponentProps<any, any>,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default defaultBehavior
+   */
+  accessibility?: Accessibility
+
+  /** Initial activeIndex value. */
+  defaultOpen?: boolean
+
+  /** Shorthand array of props for sub tree. */
+  items?: ShorthandValue[]
+
+  /**
+   * Called when a tree title is clicked.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All title props.
+   */
+  onItemClick?: ComponentEventHandler<TreeListItemProps>
```

Comment:
this one can be safely removed, as client will be able to utilize `title` shorthand's API for that